column-profiling/enclave/src/main/java/com/r3/conclave/sample/enclave/DAEnclave.java

```diff
@@ -0,0 +1,87 @@
+package com.r3.conclave.sample.enclave;
+
+import com.r3.conclave.enclave.Enclave;
+import com.r3.conclave.enclave.EnclavePostOffice;
+import com.r3.conclave.mail.EnclaveMail;
+import com.r3.conclave.sample.dataanalysis.common.DataSet;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.r3.conclave.sample.dataanalysis.common.DataSetSerializer;
+import com.r3.conclave.sample.dataanalysis.common.UserProfile;
+import java.io.ByteArrayInputStream;
+import java.util.*;
+
+
+// Return the frequency distribution of values in Dataset
+public class DAEnclave extends Enclave {
+    private Kryo kryo = new Kryo();
+
+
+    private static String dataAnalysis(ArrayList<UserProfile> arr) {
+        Map<Integer, Integer> ageFreqDist = new HashMap<>();
+        Map<String, Integer> countryFrqDist = new HashMap<>();
+        Map<String, Integer> genderFreqDist = new HashMap<>();
+
+        for( UserProfile user: arr){
+            ageFreqDist.put(user.getAge(), ageFreqDist.get(user.getAge()) == null ? 1: ageFreqDist.get(user.getAge())+1);
+            countryFrqDist.put(user.getCountry(), countryFrqDist.get(user.getCountry()) == null ? 1: countryFrqDist.get(user.getCountry())+1);
```

Comment:
You might want to use `Map.merge`, but it's up to you if you think it will make the code easier to understand or not. https://blog.frankel.ch/map-merge-compute/

Reply:
This is good catch. I have added Map.merge.